File: src/main/java/com/tournament/app/footycup/backend/dto/format/bracket/BracketNodeResponse.java
package com.tournament.app.footycup.backend.dto.format.bracket;

import com.tournament.app.footycup.backend.dto.common.MatchRef;

public record BracketNodeResponse(
        Long id,
        Integer round,
        Integer position,
        Long parentHomeNodeId,
        Long parentAwayNodeId,
        MatchRef match
) {
}

File: frontend/src/app/models/format/bracket/bracket-node.response.ts
import { MatchRef } from "../../common/match-ref.model";

export interface BracketNodeResponse {
    id: number;
    round: number;
    position: number;
    parentHomeNodeId: number | null;
    parentAwayNodeId: number | null;
    match: MatchRef | null;
}